packages/front-end/components/Radix/Button.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button as RadixButton, ButtonProps, Text } from "@radix-ui/themes";
+import { CSSProperties, ReactNode } from "react";
+import { Responsive } from "@radix-ui/themes/dist/cjs/props";
+import Link from "next/link";
+import ConditionalWrapper from "@/components/ConditionalWrapper";
+
+type Overwrite<T, NewT> = Omit<T, keyof NewT> & NewT;
+
+export type Color = "primary" | "danger";
+export type Variant = "solid" | "soft" | "outline" | "ghost";
+export type Size = "xs" | "sm" | "md";
+
+export type Props = Overwrite<
+  ButtonProps,
+  {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onClick?: (e: any) => void;
+    href?: string;
+    color?: Color | ButtonProps["color"];
+    variant?: Variant;
+    size?: Size;
+    disabled?: boolean;
+    loading?: boolean;
+    children: ReactNode;
```

Comment:
I would restrict this to string only instead of `ReactNode` to discourage deviating from the design system.  We will definitely want to support left/right icons, but I would do those as dedicated props instead.